v3-sdk/trading/src/libs/trading.ts

```diff
@@ -50,71 +70,121 @@ export async function createTrade(): Promise<TokenTrade> {
       fromReadableAmount(
         CurrentConfig.tokens.amountIn,
         CurrentConfig.tokens.in.decimals
-      )
+      ).toString()
     ),
     outputAmount: CurrencyAmount.fromRawAmount(
       CurrentConfig.tokens.out,
-      amountOut
+      JSBI.BigInt(amountOut)
     ),
     tradeType: TradeType.EXACT_INPUT,
   })
 
+  console.log('Unchecked Trade', uncheckedTrade)
+
   return uncheckedTrade
 }
 
 export async function executeTrade(
   trade: TokenTrade
 ): Promise<TransactionState> {
   const walletAddress = getWalletAddress()
-  if (!walletAddress) {
+  const provider = getProvider()
+
+  if (!walletAddress || !provider) {
     throw new Error('Cannot execute a trade without a connected wallet')
   }
 
   const options: SwapOptions = {
     slippageTolerance: new Percent(500, 10000), // 50 bips, or 0.50%
-    deadline: Math.floor(Date.now() / 1000) + 60 * 20, // 20 minutes from the current Unix time
+    deadline: Math.floor(Date.now() / 1000) + 60 * 30, // 20 minutes from the current Unix time
     recipient: walletAddress,
   }
 
   const methodParameters = SwapRouter.swapCallParameters([trade], options)
 
-  const res = await sendTransaction({
+  const tokenApproval = await getTokenTransferApprovals(
+    provider,
+    CurrentConfig.tokens.in.address,
+    walletAddress
+  )
+
+  // Fail if transfer approvals do not go through
+  if (tokenApproval !== TransactionState.Sent) {
+    return TransactionState.Failed
+  }
```

Comment:
Is this needed needed in addition to the address change?

Reply:
Yes! Required to swap the tokens and probably why the routing guide is failing